graaljs/graaljs-maven-webpack-qrcode/README.md

```diff
@@ -0,0 +1,330 @@
+            "fs": false    // Exclude this module altogether
+        },
+    },
+    plugins: [
+        new EnvironmentPlugin({
+            NODE_DEBUG: false, // Set process.env.NODE_DEBUG to false
+        }),
+    ],
+};
+module.exports = () => config;
+```
+
+Create `main.mjs`, the entry point of the bundle, with the following contents:
+```js
+// GraalJS doesn't have built-in TextEncoder support yet. It's easy to import it from a polyfill in the meantime.
+import 'fast-text-encoding';
+
+// Re-export the "qrcode" module as a "QRCode" object in the exports of the bundle.
+export * as QRCode from 'qrcode';
+```
+
+## 2.4 Using the JavaScript library from Java
+
+After reading the [qrcode](https://www.npmjs.com/package/qrcode) docs, we can write Java interfaces that match the [JavaScript types](https://www.npmjs.com/package/@types/qrcode) we want to use and methods we want to call on them.
+GraalJS makes it easy to access JavaScript objects via these interfaces.
+Java method names are mapped directly to JavaScript function and method names.
+The names of the interfaces can be chosen freely, but it makes sense to base them on the JavaScript types.
+
+`src/main/java/com/example/QRCode.java`
+```java
+package com.example;
+
+interface QRCode {
+    Promise toString(String data);
+}
+```
+
+`src/main/java/com/example/Promise.java`
+```java
+package com.example;
+
+public interface Promise {
+    Promise then(ValueConsumer onResolve);
+
+    Promise then(ValueConsumer onResolve, ValueConsumer onReject);
+}
+```
+
+`src/main/java/com/example/ValueConsumer.java`
+```java
+package com.example;
+
+import java.util.function.*;
+import org.graalvm.polyglot.*;
+
+@FunctionalInterface
+public interface ValueConsumer extends Consumer<Value> {
+    @Override
+    void accept(Value value);
+}
+```
+
+Using the `Context` class and these interfaces, we can now create QR codes and convert them to a Unicode string representation or an image.
+Our example just prints the QR code to stdout.
+
+`src/main/java/com/example/App.java`
+```java
+package com.example;
+
+import java.io.*;
+import org.graalvm.polyglot.*;
+import org.graalvm.polyglot.proxy.*;
+
+public class App {
+    public static void main(String[] args) throws Exception {
+        try (Context context = Context.newBuilder("js")
+                    .allowHostAccess(HostAccess.ALL)
+                    .option("engine.WarnInterpreterOnly", "false")
+                    .option("js.esm-eval-returns-exports", "true")
+                    .option("js.unhandled-rejections", "throw")
+                    .build()) {
+            Source bundleSrc = Source.newBuilder("js", App.class.getResource("/bundle/bundle.mjs")).build(); // ①
+            Value exports = context.eval(bundleSrc);
+            QRCode qrCode = exports.getMember("QRCode").as(QRCode.class); // ②
+            String input = argv.length > 0 ? argv[0] : "https://www.graalvm.org/javascript/";
```

Comment:
`argv` -> `args` (or vice-versa in the declaration of the parameter of the method)